misc-scripts/xref_mapping/XrefParser/RefSeqGPFFParser.pm

```diff
@@ -17,340 +17,330 @@ limitations under the License.
-      $pred_mrna_source_id, $pred_ncrna_source_id,
-      $mrna_source_id, $ncrna_source_id,
-      $pred_peptide_source_id, $peptide_source_id,
-      $entrez_source_id, $wiki_source_id, $add_dependent_xref_sth,
-      $species_id, $type, $refseq_ids,$entrez_ids,$wiki_ids
-) = @_;
-    chomp $entry;
-
-    my ($species) = $entry =~ /\s+ORGANISM\s+(.*)\n/;
-    $species = lc $species;
-    $species =~ s/^\s*//g;
-    $species =~ s/\s*\(.+\)//; # Ditch anything in parens
-    $species =~ s/\s+/_/g;
-    $species =~ s/\n//g;
-    my $species_id_check = $name2species_id->{$species};
-
-    # Try going through the taxon ID if species check didn't work.
-    if ( !defined $species_id_check ) {
-        my ($taxon_id) = $entry =~ /db_xref="taxon:(\d+)"/;
-        $species_id_check = $taxonomy2species_id->{$taxon_id};
-    }
 
-    # skip xrefs for species that aren't in the species table
-    if (   defined $species_id
-        && defined $species_id_check
-        && $species_id == $species_id_check )
-    {
-      my $xref = {};
-      my ($acc) = $entry =~ /ACCESSION\s+(\S+)/;
-      my ($ver) = $entry =~ /VERSION\s+(\S+)/;
-      my ($refseq_pair) = $entry =~ /DBSOURCE\s+REFSEQ: accession (\S+)/;
-
-      # get the right source ID based on $type and whether this is predicted (X*) or not
-      my $source_id;
-      if ($type =~ /dna/) {
-	if ($acc =~ /^XM_/ ){
-	  $source_id = $pred_mrna_source_id;
-	} elsif( $acc =~ /^XR/) {
-	  $source_id = $pred_ncrna_source_id;
-	} elsif( $acc =~ /^NM/) {
-	  $source_id = $mrna_source_id;
-	} elsif( $acc =~ /^NR/) {
-	  $source_id = $ncrna_source_id;
-	}
-      } 
-      elsif ($type =~ /peptide/) {
-	if ($acc =~ /^XP_/) {
-	  $source_id = $pred_peptide_source_id;
-	} else {
-	  $source_id = $peptide_source_id;
-	}
-      }
-      print "Warning: can't get source ID for $type $acc\n" if (!$source_id);
-
-      # Description - may be multi-line
-      my ($description) = $entry =~ /DEFINITION\s+([^[]+)/s;
-      print $entry if (length($description) == 0);
-      $description =~ s/\nACCESSION.*//s;
-      $description =~ s/\n//g;
-      $description =~ s/\s+/ /g;
-      $description = substr($description, 0, 255) if (length($description) > 255);
-
-      my ($seq) = $entry =~ /^\s*ORIGIN\s+(.+)/ms; # /s allows . to match newline
-      my @seq_lines = split /\n/, $seq;
-      my $parsed_seq = "";
-      foreach my $x (@seq_lines) {
-        my ($seq_only) = $x =~ /^\s*\d+\s+(.*)$/;
-        next if (!defined $seq_only);
-        $parsed_seq .= $seq_only;
-      }
-      $parsed_seq =~ s#//##g;    # remove trailing end-of-record character
-      $parsed_seq =~ s#\s##g;    # remove whitespace
+  my @coded_by_list = $genbank_rec =~ /\/coded_by=\"(.*?):/xg;
+
+  my $coded_by = pop @coded_by_list;
+
+  $xref->{PAIR} = $coded_by if defined $coded_by;
+
+  my ($refseq_pair) = $genbank_rec =~ /DBSOURCE\s+REFSEQ: accession (\S+)/x;
+  if (!exists $xref->{PAIR}) {
```

Comment:
Why not combine the conditions with '&&'  (&& defined $refseq_pair)

Reply:
done